server/routes/replayEventRouter.js

```diff
@@ -0,0 +1,57 @@
+const express = require('express');
+const replayEventRouter = express.Router();
+
+const { handleEvent, getDomainEvents, dispatch } = require('../models/DomainEvent');
+
+const Session = require('../infra/database/Session');
+const { publishMessage } = require('../infra/messaging/RabbitMQMessaging');
+
+const { EventRepository } = require('../infra/database/PgRepositories');
+const { handleVerifyCaptureProcessed } = require('../services/EventHandlers');
+
+replayEventRouter.post("/", async function(req, res) {
+    const session = new Session(false);
+    const eventRepository = new EventRepository(session);
+    const executeGetDomainEvents = getDomainEvents(eventRepository);
+    const domainEvents = await executeGetDomainEvents({"status":req.body.status});
+    if(req.body.status == "raised"){
+        const eventDispatch = dispatch(eventRepository, publishMessage);
+        try {
+            domainEvents.forEach(async (domainEvent) => {
+                const result = await executeGetDomainEvents({"id":domainEvent.id});
+                eventDispatch(domainEvent);
+            });
+        } catch(e) {
+            console.log(e);
+            if (session.isTransactionInProgress()){
+                await session.rollbackTransaction();
+            }
+            if (migrationSession.isTransactionInProgress()) {
+               await migrationSession.rollbackTransaction();
+            }
+            let result = e;
+            res.status(422).json({...result[0]});
+        }
+    }else if(req.body.status == "received"){
+        const eventHandle = handleEvent(handleVerifyCaptureProcessed);
+        domainEvents.forEach(async (domainEvent) => {
+            try {
+                eventHandle(domainEvent.payload);
+            } catch(e) {
+                console.log(e);
+                if (session.isTransactionInProgress()){
+                    await session.rollbackTransaction();
+                }
+                if (migrationSession.isTransactionInProgress()) {
+                    await migrationSession.rollbackTransaction();
+                }
+                let result = e;
+                res.status(422).json({...result[0]});

```

Comment:
While processing an event, if we encounter an error, no need to respond right away with 422. Instead log the error in the console and proceed to the next event. At the very end we could respond with a 207 code if there were any errors. If no errors, then we could respond with 200.